CMakeLists.txt

```diff
@@ -43,6 +43,10 @@ if(NOT MSVC)
   endif(USE_CCACHE)
 endif(NOT MSVC)
 
+if(USE_ALTERNATE_LINKER)
```

Comment:
Is there a strong reason for this option? It might be better to leave it for the user to configure custom linker, if they wish (they can always do it themselves).

One main reason is that we want the cmake to be minimal and robust, that can still against compiler API changes, because maintainability is one of important concern.

Reply:
It's optional, but we've added it in GDAL and QGIS.

It can speed up the build.